### ISSUE TITLE
karma: ReferenceError: Can't find variable: require

### ISSUE THREAD

**Issue by marcucio** [mozilla/localForage#365]
When I am running my qunit tests using karma I am getting this error when loading localforage library:

ReferenceError: Can't find variable: require

The issue is that qunit defines "module" and "module.exports" so this check we have passes (but require is undefined)

```
 if (typeof module !== 'undefined' && module.exports)
```

I have a fix for it, changing it to:

```
if (typeof module !== 'undefined' && module.exports && typeof require !== 'undefined')
```

fixes the issue. There might be a different fix when I set up karma or qunit but I didn't find one, and it might be good to check for require before using it.

I will submit a pull request for this bug in a few minutes

**Comment by glade-at-gigwell** [comment]
Is this pull request getting any movement?  We found several copies of this line in localforage.js that need the modification in order to run under karma.

**Comment by tofumatt** [comment]
I don't actually see a related pull request. If you could submit the code/PR that'd be great; I can take a look soon :smile:

**Comment by marcucio** [comment]
I's here:
https://github.com/mozilla/localForage/pull/366

**Comment by tofumatt** [comment]
Oops, sorry, I’m dumb. Looking now!

-tofumatt

On 13 May 2015 at 17:07:59, Mike (notifications@github.com) wrote:

I's here:
#366

**Comment by tofumatt** [comment]
Fixed in #378.
